### DISCUSSION TITLE
How to do filter or search on JSONField()?

### DISCUSSION THREAD

**Discussion by elcolie** [graphql-python/graphene-django#1164] [Q&A] [answered]
The original is [here](https://stackoverflow.com/questions/56902395/need-guide-lines-or-hints-to-use-graphene-python-django-filters-query-on-js)

If you know the way how to do that please share

**Comment by zbyte64** [comment, marked as answer]
Relevant django-filter issues: https://github.com/carltongibson/django-filter/issues/583 https://github.com/carltongibson/django-filter/issues/426

You can do this by manually defining your own filter, example from the first issue:
```
MarkaMoto = django_filters.CharFilter(name='attributes', lookup_expr='MarkaMoto__value__contains')
```

But there are limitations like not supporting `icontains` or querying arrays.

**Comment by elcolie** [comment]
Thank you very much @zbyte64 I though `graphene-django` has feature to do a generic filter for this problem since it has been around for a year already.

**Reply by zbyte64** [reply]
We aren't allowed to have a graphql endpoint with arbitrary keys. For generic filtering, one would have to represent arbitrary key value pairs as it's own input type. The problem looks fairly similar to generic filtering of models across arbitrary columns and has the same concerns.
